### ISSUE TITLE
move the sudo library outside libdir

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#1987]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/1983
- Created at 2013-06-14 09:34:20 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:11:45 as wontfix
- Assigned to [mmsrubar](https://pagure.io/user/mmsrubar)
---

For historical reasons the libsss_sudo library is packaged in libdir and that's also where the sudo binary expects it and dlopens it from.

But it's wrong from packaging perspective as the library cannot be linked with. We should move it to the same location as libsss_autofs. This change would require careful coordination with sudo at packaging level to make sure sudo'd dlopen path is changed as well.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2013-06-20 15:42:24](https://pagure.io/SSSD/sssd/issue/1983#239625)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.12 beta
rhbz:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-01-30 17:31:00](https://pagure.io/SSSD/sssd/issue/1983#239626)**

The fix in SSSD involves changing the default value of where we install the sudo module from /usr/lib to whatever `appmodpath` was set to.

The second part of the fix would be to change the default value of where the sudo plugin looks for the module. They seem to have an option called `--with-sssd-lib` as well.

owner: somebody => mmsrubar

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2014-09-26 18:49:57](https://pagure.io/SSSD/sssd/issue/1983#239627)**

Fields changed

mark:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-02-10 14:25:21](https://pagure.io/SSSD/sssd/issue/1983#239628)**

Fields changed

patch: 0 => 1

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2015-06-25 15:35:51](https://pagure.io/SSSD/sssd/issue/1983#239629)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog
sensitive:  => 0

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-01-11 14:48:21](https://pagure.io/SSSD/sssd/issue/1983#239630)**

Fields changed

blockedby:  => #2919

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-02-16 15:30:52](https://pagure.io/SSSD/sssd/issue/1983#239631)**

This depends on refactoring the sudo protocol first, which is a 1.14 beta task.

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-06-20 11:32:43](https://pagure.io/SSSD/sssd/issue/1983#239632)**

This didn't happen in the 1.14 milestone again, but since it's "only" a packaging violation, it's not terribly critical.

milestone: SSSD 1.14 beta => SSSD 1.15 beta

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 15:00:35](https://pagure.io/SSSD/sssd/issue/1983#239633)**

**Metadata Update from @jhrozek**:
- Issue assigned to mmsrubar
- Issue marked as depending on: #2919
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-13 14:19:36](https://pagure.io/SSSD/sssd/issue/1983#633180)**

**Metadata Update from @thalman**:
- Custom field blocking adjusted to 2919
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch adjusted to on (was: 1)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: Canditate to close

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:11:44](https://pagure.io/SSSD/sssd/issue/1983#636656)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:11:45](https://pagure.io/SSSD/sssd/issue/1983#636657)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
